src/lib.rs

```diff
@@ -185,15 +166,15 @@ unsafe fn decryptbuf(dest: &mut [u8], src: usize) {
 #[derive(Copy, Clone)]
 #[repr(transparent)]
 pub struct ObfBuffer<A>(A);
-impl<A: FixedSizeArray<u8>> ObfBuffer<A> {
+impl<A: AsRef<[u8]>> ObfBuffer<A> {
 	#[allow(deprecated)]
 	unsafe fn uninit() -> Self {
 		mem::uninitialized()
```

Comment:
A bit of a future enhancement: remove this function, and create a `MaybeUninit<ObfBuffer>` where necessary instead? I'll see if I can make it work.

Reply:
I don't think this is possible. The only interface available to get the underlying buffer is `AsRef` and `AsMut` traits which don't work with MaybeUninit types to get their underlying buffers.

Reply:
https://github.com/CasualX/obfstr/pull/24 I mean, it's possible this way. It doesn't change the behavior of the existing code and still has the problem that passing a Vec<u8> here will trigger UB. But it at least avoids mem::uninitialized, which is unsound and could break our code at any moment.

Reply:
If it's okay I've chosen the clone() method. See the other resolved comment for more info.